src/OpenLoco/Map/TileManager.cpp

```diff
@@ -74,7 +77,7 @@ namespace OpenLoco::Map::TileManager
     {
         TileHeight height{ 16, 0 };
         // Off the map
-        if ((unsigned)pos.x >= 12287 || (unsigned)pos.y >= 12287)
+        if ((unsigned)pos.x >= (Map::map_width - 1) || (unsigned)pos.y >= (Map::map_size - 1))
```

Comment:
```suggestion
        if ((unsigned)pos.x >= (Map::map_width - 1) || (unsigned)pos.y >= (Map::map_height - 1))
```

Reply:
Are we ever actually going to have none square maps? Perhaps width/height never makes sense.

Reply:
I'm not sure why height is correct here? The map is still three dimensional, this is the tiles going on x y axis, z is height.

Reply:
It might be worth having better constants for this that are actually defined as map_pos. We should also define the lower bounds of such coordinates not just the upper.
Example would be:
```
auto MapMin = map_pos{ 0, 0, 0 };
auto MapMax = map_pos{ MapSizeX * TileSize, MapSizeY * TileSize, *** };
```
This clearly specifies what the valid value range is. Functions like ```isPosValid(map_pos(3)&)``` should rely on that. At some point we should probably also add ```Bounds``` which just stores two vectors for min, max, this helps with things like ```Bounds b{a, b}; b.isWithin(pos); b.intersects(other);``` etc.

Also defining the lower bounds allows to change the map pivot if we like to, so we could have {0,0,0} at the map center knowing the lower bounds we can translate the position always back into the index.

Reply:
map_size was indeed the wrong one. @duncanspumpkin suggested we rename them at some point to actually reflect width, depth, height.